File: 프로그래머스/lv2/17677. ［1차］ 뉴스 클러스터링/［1차］ 뉴스 클러스터링.py
from collections import Counter

def solution(str1, str2):
    answer = 0
    low_str1 = str1.lower()
    low_str2 = str2.lower()
    
    str1_lst = []
    str2_lst = []
    
    for i in range(len(low_str1) - 1):
        if low_str1[i].isalpha() and low_str1[i+1].isalpha():
            str1_lst.append(low_str1[i:i+2])

    for i in range(len(low_str2) - 1):
        if low_str2[i].isalpha() and low_str2[i+1].isalpha():
            str2_lst.append(low_str2[i:i+2])
                   
    intersection = list((Counter(str1_lst) & Counter(str2_lst)).elements())
    union = list((Counter(str1_lst) | Counter(str2_lst)).elements())
                   
    if len(intersection) == 0 and len(union) == 0:
        return 65536
    else:
        return int(len(intersection) / len(union) * 65536)
    

    
    return answer

File: 프로그래머스/lv2/17677. ［1차］ 뉴스 클러스터링/［1차］ 뉴스 클러스터링.java
import java.util.*;

class Solution {
    public int solution(String str1, String str2) {
        str1 = str1.toLowerCase();
        str2 = str2.toLowerCase();
        
        ArrayList<String> lst1 = new ArrayList<>();
        ArrayList<String> lst2 = new ArrayList<>();
        
        ArrayList<String> intersection = new ArrayList<>();
        ArrayList<String> union = new ArrayList<>();
        
        for(int i = 0; i < str1.length() - 1; i++){
            char a = str1.charAt(i);
            char b = str1.charAt(i+1);
            
            if (Character.isLetter(a) && Character.isLetter(b)) {
                lst1.add(a + "" + b);
            }
        }
        
        for(int i = 0; i < str2.length() - 1; i++){
            char a = str2.charAt(i);
            char b = str2.charAt(i+1);
            
            if(Character.isLetter(a) && Character.isLetter(b)){
                lst2.add(a + "" + b);
            }
        }
        
        Collections.sort(lst1);
        Collections.sort(lst2);
        
        for(String s : lst1){
            if(lst2.remove(s)){
                intersection.add(s);
            }
            union.add(s);
        }
        
        for(String s : lst2) {
            union.add(s);
        }
        
        double answer = 0;
        if(union.size() == 0){
            answer = 65536;
        } else {
            answer = (double)intersection.size() / (double)union.size() * 65536;
        }
        
        return (int)answer;
    }
}